replica/table.cc

```diff
@@ -542,26 +543,89 @@ void table::enable_off_strategy_trigger() {
     do_update_off_strategy_trigger();
 }
 
-compaction_group_vector table::make_compaction_groups() {
-    compaction_group_vector ret;
-    auto&& ranges = dht::split_token_range_msb(_x_log2_compaction_groups);
-    ret.reserve(ranges.size());
-    tlogger.debug("Created {} compaction groups for {}.{}", ranges.size(), _schema->ks_name(), _schema->cf_name());
-    size_t i = 0;
-    for (auto&& range : ranges) {
-        ret.emplace_back(std::make_unique<compaction_group>(*this, i++, std::move(range)));
+class single_compaction_group_manager final : public compaction_group_manager {
+    replica::table& _t;
+public:
+    single_compaction_group_manager(replica::table& t) : _t(t) {}
+
+    compaction_group_vector make_compaction_groups() const override {
+        compaction_group_vector r;
+        r.push_back(std::make_unique<compaction_group>(_t, size_t(0), dht::token_range::make_open_ended_both_sides()));
+        return r;
     }
-    return ret;
+    size_t compaction_group_of(dht::token) const override {
+        return 0;
+    }
+    size_t log2_compaction_groups() const override {
+        return 0;
+    }
+};
+
+class tablet_compaction_group_manager final : public compaction_group_manager {
```

Comment:
Isn't single_compaction_group_manager the same as this one, with just one compaction group?

Reply:
I guess the discovery is different.